### PR TITLE
chore: prettier, tslint and cleanup

### PR DESCRIPTION
**Pull request by zobo**
Add warning message on windows platform and do not start the language server.
Change compile directory to standard "out".
Update tslint and add prettier.
Reformat all files.

### CODE REVIEW

**Comment by zobo** [comment]
Still a work in progress.

**Comment by dantleech** [comment]
thanks, this is much needed! do  you think we could make separate PRs?

- PR for adding prettier / TSLint (which are the majority of the changes here)
- PR for adding the warning etc

**Comment by zobo** [comment]
Yes, I agree, as this got a bit out of hand last night. I will clean it up and make a series of PRs.
1. tsling, prettier, eslint
2. warning mesage
3. Chane to how phpactor LS is bundled
4. Change of publisher and GH actions

Please leave this open and I will try to refactor it today when I have time.

**Comment by zobo** [comment]
I think this PR is ok now. Let me know if you see anything you don't like.
I'm looking into the windows warning again. There is a concept of [when in the extension manifest](https://code.visualstudio.com/api/references/when-clause-contexts) that could perhaps be used to disable the extension on windows - but I can't seem to get it working. In either case, all the registered command also need to be disabled. I'll do a separate PR for that. After that comes the repackage of phpactor LS. And lastly what is needed to publish.

File: .github/workflows/semantic.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: 'Semantic Pull Request'
```
Comment (dantleech) [inline .github/workflows/semantic.yml:1]: what does this do?

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "version": "2.0.0",
```
Comment (dantleech) [inline .vscode/tasks.json:2]: what does this do?

**Review by dantleech — APPROVED** [review]

**Comment by dantleech** [comment]
looks good from a quick glance :+1: a few questions above

File: .vscode/tasks.json (hunk 2 above)
Comment (zobo) [reply .vscode/tasks.json:2]: Files for better developer experience. If you develop this extension in vscode you can easily start nom watch with a shortcut. I can add some info to README the same way I have in vscode-php-debug.

File: .github/workflows/semantic.yml (hunk 1 above)
Comment (zobo) [reply .github/workflows/semantic.yml:1]: Semantic pull requests and semantic versioning. I'll write a detailed explanation once I'm back. We should probably first discuss release procedures.

Comment (zobo) [reply .github/workflows/semantic.yml]: I've removed the file from this branch and we'll deal with it in the next PR.

File: README.md [hunk 3]
```diff
@@ -27,16 +25,14 @@ For development it is easier
...
 
 - `phpactor.update`: Update Phpactor to the latest version
 - `phpactor.status`: Show Phpactor's status
 - `phpactor.reindex`: Reindex the project.
 - `phpactor.services.list`: List Phpactor's currently running services.
 - `phpactor.config.dump`: Dump Phpactor's configuratoin to the log window.
 
-Documentation
--------------
+## Documentation
```
Comment (dantleech) [inline README.md:36]: assume these were linter changes, i prefer `---` but hey, not important :)

**Review by dantleech — APPROVED** [review]

**Comment by dantleech** [comment]
:+1: thanks!

**Pull request merged**